services/bitbucket/bitbucket-pipelines.service.js

```diff
@@ -70,9 +76,22 @@ module.exports = class BitbucketPipelines extends BaseJsonService {
   }
 
   async fetch({ user, repo, branch }) {
-    const url = `https://api.bitbucket.org/2.0/repositories/${user}/${repo}/pipelines/`
+    let branchName
+    if (branch == null) {
+      const repoUrl = `https://api.bitbucket.org/2.0/repositories/${user}/${repo}/`
+      const repoResp = await this._requestJson({
+        url: repoUrl,
+        schema: bitbucketRepoSchema,
+        errorMessages: { 403: 'private repo' },
+      })
```

Comment:
Completely orthogonal to this PR, but just hit me we support private repos with other BB badges in the self-hosted Shields scenario, and may be worth doing so here at some point in the future (though admittedly I don't think anyone's asked for it)

Reply:
Yes there are several inconsistencies within the BB service family. For example the Pull Requests badge works with both BB Cloud and BB Server, but Issues and Pipelines only work with BB Cloud. There's also a mix of V1 and V2 API calls. I don't think anything in we are doing in this PR prevents us from sorting these out at another time.